### ISSUE TITLE
dio: ^3.0.10 - Post Data Not Work

### ISSUE THREAD

**Issue by gushendra86** [flutterchina/dio#951]
### New Issue Checklist

- [v]  I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |   flutter   |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |   1.20.4   |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |   ^3.0.10   |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |   100%   |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

I write this code on Flutter and the server cannot receive the post data, always get null. But if i try to send data via postman or any rest api client, i get the data. This only happen on dio version ^3.0.10.
If i use dio version 3.0.9, the post data is sent successfully.

Flutter Code :
```@override
  void initState() {
    getHttp();
    super.initState();
  }

  void getHttp() async {
    try {
      Response response;
      Dio dio = new Dio();
      response = await dio.post("http://192.168.0.4/api/test.php", data: {"param1": "satu", "param2": 2, "param3": 10});
      print(response.data.toString());
    } catch (e) {
      print(e);
    }
  }
```

And this is the php script :
```
<?php
if(isset($_POST['param1']) && !empty($_POST['param1'])){
	echo $_POST['param1'];
} else {
	echo 'null';
}
echo '<br>';

if(isset($_POST['param2']) && !empty($_POST['param2'])){
	echo $_POST['param2'];
} else {
	echo 'null';
}
echo '<br>';

if(isset($_POST['param3']) && !empty($_POST['param3'])){
	echo $_POST['param3'];
} else {
	echo 'null';
}
```
This is the screenshoot of code using dio 3.0.10
![image](https://user-images.githubusercontent.com/62778692/94646821-503a5d00-0319-11eb-894f-a979d4901635.png)

This is the screenshoot using rest api client
![image](https://user-images.githubusercontent.com/62778692/94646882-75c76680-0319-11eb-9f91-bbb6863a9a98.png)

Thank you and regards,

Hendra

**Comment by IkhwanSI13** [comment]
in my case, if I use the POST method with the data parameter, it works. But if I use it with the data parameter and the option parameter, it will empty the data parameter.

Dio 3.0.10
Flutter (Channel stable, 1.20.4)

**Comment by beblessed2009** [comment]
wrap post data with FormData like this:
await instance()
        .post(apiUrl, data: FormData.fromMap(formData), options: _options)
        .then((resp) => response = resp)
        .catchError((e) {});

**Comment by HelgeSverre** [comment]
Can verify that this happens on 3.0.10, the input is emptied.
Reverting to 3.0.9 fixes the problem.
